### DISCUSSION TITLE
Getting starterd with openTDF

### DISCUSSION THREAD

**Discussion by Vivekmauli14** [opentdf/platform#878] [Platform Q&A] [answered]
Hello,

I am trying to set up OpenTDF on my Ubuntu server. I have followed the installation steps, but I am encountering an error when I run the command `go run ./service start`. 

Here is the relevant part of the output:

![screenshot_2024-05-28_154747_720](https://github.com/opentdf/platform/assets/83024029/1a571db0-0dbc-43f3-a4f3-1cde05dd4ef3)

I have ensured that Go is installed and set up correctly. I also checked that all necessary dependencies are in place. Do I need to configure any API tokens or additional configuration files? Any guidance or help would be greatly appreciated!

Thank you!

**Comment by pacog1994** [comment]
I don't think this is well-documented or if this will fix your issue but try running `go run ./service provision keycloak`.

**Reply by jrschumacher** [reply]
You're right, you always need to provision keycloak before running OpenTDF locally. This is because OpenTDF depends on an OIDC compliant IdP to function. This is mapped in the [example configuration](https://github.com/opentdf/platform/blob/main/opentdf-example.yaml#L33-L34).

If you had an existing IdP running, you could point OpenTDF to it and configure the Issuer to effectively say "I trust tokens from this **issuer** as long as I am the **audience**."

We are expecting to spend more time on documentation this month, so hopefully we can make this clearer. **Thanks for helping to support the community!**

**Comment by jrschumacher** [comment]
@Vivekmauli14 are these the steps you performed?

1. `cp opentdf-example.yaml opentdf.yaml`
2. `docker compose -f docker-compose.yaml up`

**Reply by Vivekmauli14** [reply]
Hi @jrschumacher,

yes I followed the README, and tried the steps you have mentioned above, I have set up the Platform and Client Web, But now I am getting error in client not found when trying to log into opentdf. I tried creating a client in opentdf realm but it is of no help and I am stuck in there.
Your guidance and help in resolving this issue is much appreciated.

Have a look at the error below:

![Screenshot 2024-06-04 110326](https://github.com/opentdf/platform/assets/83024029/b4811bde-cc65-477c-8213-94265bf461e5)

**Reply by jrschumacher** [reply, marked as answer]
@Vivekmauli14 the issue here is that the `opentdf` realm doesn't exist in Keycloak. You might want to authenticate at `http://localhost:8888/auth` and see if you can see it under the dropdown.

If not then you'll need to provision keycloak either manually or via the platform command. I think this will give you more info `go run ./service provision --help`.
